src/java.base/share/classes/javax/net/ssl/SSLParameters.java

```diff
@@ -696,4 +697,76 @@ public void setApplicationProtocols(String[] protocols) {
         }
         applicationProtocols = tempProtocols;
     }
+
+    /**
+     * Returns a prioritized array of signature scheme names that can be used
+     * over the SSL/TLS/DTLS protocols.
+     * <p>
+     * The returned array includes {@code String} names from the list of
+     * standard signature schemes in the <a href=
+     * "{@docRoot}/../specs/security/standard-names.html#signature-schemes">
+     * Signature Schemes</a> section of the Java Cryptography
+     * Architecture Standard Algorithm Name Documentation, and may also
+     * include other signature schemes that the provider supports.
```

Comment:
There doesn't seem to be anything preventing a user from setting a bogus signature scheme (ex: named "foo") - which is neither a standard name or a provider specific name.
I think the method may be too tightly specified, and you should make this more general and not put any constraints on the names of the signature schemes. (Although we should still link to the specification for a list of standard names). 

It would be useful to explain when this method returns pre-populated scheme names as supported by the underlying provider and when it may return an empty list.

Reply:
You should also define the interaction with the system properties (probably as an @implNote). Does `getSignatureSchemes` ever return the value of the system properties? Does the API always override the properties?

Reply:
I asked a similar question about the System property interaction in the CSR and he has some answers there.

Reply:
All good catches, Sean and Jamil.  I updated the CSR accordingly, and will update the PR after I complete other comments.